spec/codeowners/cli/owners_list_handler_spec.rb

```diff
@@ -0,0 +1,202 @@
+    let(:new_file) { 'text.rb' }
+    let(:the_owner) { '@owner1' }
+    let(:another_owner1) { '@owner2' }
+    let(:another_owner2) { '@owner3' }
+    let(:line_str) { "#{new_file} #{the_owner}" }
+    let(:line) { Codeowners::Checker::Group::Line.build(line_str) }
+    let(:suggestion) { <<~QUESTION }
+      Unknown owner: #{the_owner} for pattern: #{new_file}. Add owner to the OWNERS file?
+      (y) yes
+      (i) ignore owner in this session
+      (q) quit and save
+    QUESTION
+    let(:suggestion_options) { %w[y i q] }
+    let(:owners_list) { instance_double(::Array, 'owners_list') }
+    let(:checker) { instance_double(Codeowners::Checker, 'checker') }
+
+    before do
+      allow(owners_list).to receive(:<<)
+      allow(checker).to receive(:owners_list).and_return(owners_list)
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
+    def self.test_with_non_ignored_owner
+      it 'suggest owner addition' do
+        allow(owners_list_handler).to receive(:ask)
+
+        owners_list_handler.suggest_add_to_owners_list(line, the_owner)
+
+        expect(owners_list_handler).to have_received(:ask).with(suggestion, limited_to: suggestion_options)
+      end
+
+      it 'updates owners list if user chooses to add' do
+        allow(owners_list_handler).to receive(:ask).and_return('y')
+        owners_list_handler.checker = checker
+
+        output = owners_list_handler.suggest_add_to_owners_list(line, the_owner)
+
+        expect(owners_list).to have_received(:<<).with(the_owner)
+        expect(owners_list_handler.content_changed).to be_truthy
+        expect(owners_list_handler.ignored_owners).not_to include(the_owner)
+        expect(output).to be_truthy
+      end
+
+      it 'update ignored owners and returns nil if user chooses to ignore' do
+        allow(owners_list_handler).to receive(:ask).and_return('i')
+        owners_list_handler.checker = checker
+
+        output = owners_list_handler.suggest_add_to_owners_list(line, the_owner)
+
+        expect(owners_list).not_to have_received(:<<)
+        expect(owners_list_handler.content_changed).not_to be_truthy
+        expect(owners_list_handler.ignored_owners).to include(the_owner)
+        expect(output).to be_nil
+      end
+
+      it 'throws :user_quit and does not change anything' do
+        allow(owners_list_handler).to receive(:ask).and_return('q')
+        owners_list_handler.checker = checker
+
+        expect { owners_list_handler.suggest_add_to_owners_list(line, the_owner) }.to throw_symbol :user_quit
+
+        expect(owners_list).not_to have_received(:<<)
+        expect(owners_list_handler.content_changed).not_to be_truthy
+        expect(owners_list_handler.ignored_owners).not_to include(the_owner)
+      end
+    end
+    # rubocop:enable Metrics/AbcSize
+    # rubocop:enable Metrics/MethodLength
+
+    # rubocop:disable RSpec/EmptyExampleGroup
+    context 'when ignored owners list is empty' do
+      before do
```

Comment:
Missed empty before.

Reply:
With `shared_example` these rubocop comments are not necessary anymore.